flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionService.java

```diff
@@ -48,9 +48,11 @@
     private final LeaderElectionDriverFactory leaderElectionDriverFactory;
 
     /** The leader contender which applies for leadership. */
+    @GuardedBy("lock")
     private volatile LeaderContender leaderContender;
```

Comment:
this is also nullable; see constructor.

Reply:
I know, I just hesitated to add this annotation because it would generate warnings in several locations within the class. But I added `@Nullable` now for the other two fields as well to make it consistent.

Reply:
urgh. I'd rather roll these 2 back :see_no_evil: